consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -122,10 +127,30 @@ where
     cmax
 }
 
+/// A cryptarchia engine that manages the block tree.
+///
+/// Whenever a new block is received, it performs the fork choice rule
+/// and update the LIB if needed (depending on [`CryptarchiaState`]).
+///
+/// This guarantees that only blocks descending from the LIB are accepted,
+/// by proactively pruning forks that are diverged before the LIB.
 #[derive(Clone, Debug)]
 pub struct Cryptarchia<Id, State: ?Sized> {
+    /// Blocks that are ancestors of the [`Self::lib`].
+    /// It is guaranteed that these blocks constitute a single, fork-less chain.
+    branches_before_lib: HashMap<Id, Branch<Id>>,
```

Comment:
why call it branches then? also, we probably can avoid this entirely, anything before the LIB can be discarded by the engine